src/databricks/labs/ucx/source_code/queries.py

```diff
@@ -43,18 +45,44 @@ def _catalog(table):
 
     def apply(self, code: str) -> str:
         new_statements = []
-        for statement in sqlglot.parse(code):
+        for statement in self.get_statements(code):
             if not statement:
                 continue
-            for old_table in statement.find_all(Table):
-                catalog = self._catalog(old_table)
-                if catalog != 'hive_metastore':
-                    continue
-                dst = self._index.get(old_table.db, old_table.name)
+            for old_table in self.get_dependencies(statement):
+                src_db = old_table.db if old_table.db else self._use_schema
+                dst = self._index.get(src_db, old_table.name)
                 if not dst:
                     continue
                 new_table = Table(catalog=dst.dst_catalog, db=dst.dst_schema, this=dst.dst_table)
                 old_table.replace(new_table)
             new_sql = statement.sql('databricks')
             new_statements.append(new_sql)
         return '; '.join(new_statements)
+
+    @staticmethod
+    def get_statements(code: str):
+        return sqlglot.parse(code)
+
+    @classmethod
+    def get_dependencies(cls, statement: Expression):
+        dependencies = []
+        for old_table in statement.find_all(Table):
+            catalog = cls._catalog(old_table)
+            if catalog != 'hive_metastore':
+                continue
+            dependencies.append(old_table)
+        return dependencies
+
+    @classmethod
+    # This method is used to get the dependencies for a query as a list of tables
+    def get_view_sql_dependencies(cls, code: str, *, use_schema: str = ""):
+        try:
+            statements = sqlglot.parse(code)
+            if len(statements) != 1 or statements[0] is None:
+                raise ValueError(f"Could not analyze view SQL: {code}")
+        except ParseError as e:
+            raise ValueError(f"Could not analyze view SQL: {code}") from e
+        statement = statements[0]
+        for table in cls.get_dependencies(statement):
+            src_db = table.db if table.db else use_schema
+            yield ucx_tables.Table("hive_metastore", src_db, table.name, "type", "")
```

Comment:
introduce new dataclass, where we only know about the database and table name. We don't know about the table type until we actually look it up in the tables crawler.